cfme/infrastructure/provider.py

```diff
@@ -159,7 +159,7 @@ def num_host(self, db=True):
                             .filter(ext_management_systems.name == self.name))
             return len(hostlist)
         else:
-            return int(self.get_detail("Relationships", "Hosts"))
+            return int(self.get_detail("Relationships", "host.png", use_icon=True))
```

Comment:
It changes to `Hosts / Nodes` when you have at least one OS Infra provider. And I was not able to reliably determine the moment when to switch so going with the path of least resistance.